require.js

```diff
@@ -87,11 +83,17 @@
     if (typeof bundle === 'object') {
       for (var key in bundle) {
         if (has.call(bundle, key)) {
-          modules[key] = bundle[key];
+          require.register(modules[key], bundle[key]);
         }
       }
     } else {
       modules[bundle] = fn;
+      if (extExp.test(bundle)) {
+        var woExt = bundle.replace(extExp, '');
+        var existing = aliases[woExt];
+        if (!existing || existing.split('.').pop() !== 'js') aliases[woExt] = bundle;
+      }
+      if (indexExp.test(bundle)) aliases[bundle.replace(indexExp, '')] = bundle;
```

Comment:
Alias `foo/bar.js` to `foo/bar`.
Alias `foo/bar/index.js` to `foo/bar/index` and `foo/bar`. Do we need more logic here so that `foo/bar/index.js` doesn't overwrite the alias for `foo/bar.js`?